src/Cli/dotnet/commands/dotnet-workload/install/NetSdkManagedInstaller.cs

```diff
@@ -139,7 +140,77 @@ public void RollBackWorkloadPackInstall(PackInfo packInfo, SdkFeatureBand sdkFea
             }
         }
 
-        public void InstallWorkloadManifest(ManifestId manifestId, ManifestVersion manifestVersion, SdkFeatureBand sdkFeatureBand) => throw new NotImplementedException();
+        public void InstallWorkloadManifest(ManifestId manifestId, ManifestVersion manifestVersion, SdkFeatureBand sdkFeatureBand)
+        {
+            string packagePath = null;
+            string tempExtractionDir = null;
+            string tempBackupDir = null;
+            var manifestPath = Path.Combine(_dotnetDir, "sdk-manifests", sdkFeatureBand.ToString(), manifestId.ToString());
+
+            _reporter.WriteLine(string.Format(LocalizableStrings.InstallingWorkloadManifest, manifestId, manifestVersion));
+
+            try
+            {
+                TransactionalAction.Run(
+                   action: () =>
+                   {
+                       packagePath = _nugetPackageDownloader.DownloadPackageAsync(WorkloadManifestUpdater.GetManifestPackageId(sdkFeatureBand, manifestId), new NuGetVersion(manifestVersion.ToString())).Result;
+                       tempExtractionDir = Path.Combine(_tempPackagesDir.Value, $"{manifestId}-{manifestVersion}-extracted");
+                       Directory.CreateDirectory(tempExtractionDir);
+                       var manifestFiles = _nugetPackageDownloader.ExtractPackageAsync(packagePath, tempExtractionDir).Result;
+
+                       if (Directory.Exists(manifestPath) && Directory.GetFileSystemEntries(manifestPath).Any())
+                       {
+                           // Backup existing manifest data for roll back purposes
+                           tempBackupDir = Path.Combine(_tempPackagesDir.Value, $"{manifestId}-{manifestVersion}-backup");
+                           FileAccessRetrier.RetryOnMoveAccessFailure(() => Directory.Move(manifestPath, tempBackupDir));
+                           Directory.Delete(manifestPath, true);
+                       }
+
+                       Directory.CreateDirectory(Path.GetDirectoryName(manifestPath));
+                       FileAccessRetrier.RetryOnMoveAccessFailure(() => Directory.Move(tempExtractionDir, manifestPath));
+                   },
+                    rollback: () => {
+                        if (!string.IsNullOrEmpty(tempBackupDir))
+                        {
+                            FileAccessRetrier.RetryOnMoveAccessFailure(() => Directory.Move(tempBackupDir, manifestPath));
+                        }
+                    });
+            }
+            catch (Exception e)
+            {
+                throw new Exception(string.Format(LocalizableStrings.FailedToInstallWorkloadManifest, manifestId, manifestVersion, e.Message));
+            }
+            finally
+            {
+                // Delete leftover dirs and files
```

Comment:
nit: this will cause a concurrent issue if there are 2 dotnet process running at the sametime. I prefer try out best to delete the empty folder. If not, the system should be able to tolerate empty folder.

Reply:
What do you mean by try our best to delete the empty folder? Move this to the try clause?

Reply:
the delete could fail, catch the exception and swallow it

Reply:
put delete in to try part instead of finally